fms/models/hf/gpt_bigcode/__init__.py

```diff
@@ -0,0 +1,91 @@
+import os
+from typing import Union
+
+from fms.models.hf.gpt_bigcode.modeling_gpt_bigcode_hf import (
+    HFAdaptedGPTBigCodeForCausalLM,
+)
+
+
+def get_model(
+    model_name_or_path: Union[str, os.PathLike]
+) -> HFAdaptedGPTBigCodeForCausalLM:
+    """
+    Get a Huggingface adapted FMS model from an equivalent HF model
+
+    Parameters
+    ----------
+    model_name_or_path: Union[str, os.PathLike]
+        Either the name of the model in huggingface hub or the absolute path to
+        the huggingface model
+
+    Returns
+    -------
+    HFAdaptedGPTBigCodeForCausalLM
+        A Huggingface adapted FMS implementation of GPT-BigCode
+    """
+    from fms.models.hf.utils import register_fms_models
+    from transformers import AutoModelForCausalLM
+    import torch
+    from fms.models.gpt_bigcode import GPTBigCode
+
+    register_fms_models()
+    hf_model = AutoModelForCausalLM.from_pretrained(model_name_or_path)
+
+    model = GPTBigCode(
+        src_vocab_size=hf_model.config.vocab_size,
+        emb_dim=hf_model.config.n_embd,
+        ln_eps=hf_model.config.layer_norm_epsilon,
+        nheads=hf_model.config.n_head,
+        nlayers=hf_model.config.n_layer,
+        hidden_grow_factor=hf_model.config.n_inner / hf_model.config.hidden_size,
+        pad_id=-1,
+        max_pos=hf_model.config.max_position_embeddings,
+    )
+
+    new_hf_sd = __rename_weights_to_fms(hf_model.transformer.state_dict())
+    model.load_state_dict(new_hf_sd, strict=False)
+    with torch.no_grad():
+        for i, layer in enumerate(hf_model.transformer.h):
+            q, k, v = layer.attn.c_attn.weight.split([2048, 128, 128], dim=0)
+            q_bias, k_bias, v_bias = layer.attn.c_attn.bias.split(
+                [2048, 128, 128], dim=0
+            )
+            model.base_model.layers[i].attn.query.weight.copy_(q)
+            model.base_model.layers[i].attn.query.bias.copy_(q_bias)
+            model.base_model.layers[i].attn.key.weight.copy_(k)
+            model.base_model.layers[i].attn.key.bias.copy_(k_bias)
+            model.base_model.layers[i].attn.value.weight.copy_(v)
+            model.base_model.layers[i].attn.value.bias.copy_(v_bias)
```

Comment:
if it's not hard, I would delete the tensors from layer after they have been copied over, as for larger granite models we might hit RAM issues

Reply:
added to latest commit